File: backend/app/schemas/recommendation.py
from pydantic import BaseModel, Field

from app.schemas.place import Recommendation
from app.schemas.user import UserProfile


class RecommendationRequest(BaseModel):
    user_profile: UserProfile
    travel_date: str | None = Field(default=None, examples=["2026-07-10"])


class RecommendationResponse(BaseModel):
    recommendations: list[Recommendation]

File: frontend/src/api/recommendation.ts
import { client } from "./client";
import type { UserProfile } from "../types/user";
import type { Recommendation } from "../types/place";

export interface RecommendationRequest {
  user_profile: UserProfile;
  travel_date?: string;
}

export interface RecommendationResponse {
  recommendations: Recommendation[];
}

export async function postRecommendations(
  payload: RecommendationRequest
): Promise<Recommendation[]> {
  const { data } = await client.post<RecommendationResponse>(
    "/api/recommendations",
    payload
  );
  return data.recommendations;
}
